### DISCUSSION TITLE
How to adjust -vm setting products?

### DISCUSSION THREAD

**Discussion by phermsdorf** [eclipse-justj/justj#35] [Q&A]
Hi,

we were hit by this issue: https://bugs.eclipse.org/bugs/show_bug.cgi?id=576526
The described workaround is to use javaw.exe instead of java.exe(or the jvm.dll) to launch the application,

We currently have a post build bash-script-hack in place which replaces entries like
```
-vm
plugins/org.eclipse.justj.openjdk.hotspot.jre.full.win32.x86_64_21.0.5.v20241023-1957/jre/bin
```
in the product.ini with
```
-vm
plugins/org.eclipse.justj.openjdk.hotspot.jre.full.win32.x86_64_21.0.5.v20241023-1957/jre/bin/javaw.exe
```
which is obviously not very elegant and bad to maintain.
Since this entry is (probably) created with a p.2inf instruction in the justj bundle we cannot overide that easily ...

Any ideas how to adjust that, maybe as part of the tycho build?

Thank you.
Bye Peter

**Comment by merks** [comment]
That's a weird problem.  Indeed that comes from the touchpoints and this is how all the Eclipse IDEs are shipped.  I have no idea how you'd override that.  😞

**Reply by phermsdorf** [reply]
I came up with the following solution:
* i split up the materialize-product and archive-product goals by binding the latter one to a later phase
* in beetween i adjust the product .ini

something like this:
```
<plugin>
	<groupId>org.eclipse.tycho</groupId>
	<artifactId>tycho-p2-director-plugin</artifactId>
	<executions>
		<execution>
			<id>materialize-product</id>
			<goals>
				<goal>materialize-products</goal>
			</goals>
		</execution>
		<execution>
			<id>archive-product</id>
			<phase>integration-test</phase>
			<goals>
				<goal>archive-products</goal>
			</goals>
		</execution>
	</executions>
</plugin>
<plugin>
	<artifactId>maven-antrun-plugin</artifactId>
	<executions>
	  <execution>
		<id>process-ini-javaw</id>
		<phase>pre-integration-test</phase>
		<configuration>
		  <target>
			<replace token= "/jre/bin" value="/jre/bin/javaw" dir="${project.build.directory}/products/">                                 
			  <include name="*/win32/win32/x86_64/*/*.ini"/>
			</replace>
		  </target>
		</configuration>
		<goals>
		  <goal>run</goal>
		</goals>
	  </execution>
	</executions>
</plugin>
```
If someone has a better idea ...

Bye Peter
